src/ngraph/ngraph_compiler.cc

```diff
@@ -463,44 +463,29 @@ void Compiler::CheckInNgraph() {
   for (const std::shared_ptr<ngraph_bridge::Node>& node : ngraph_.nodes_) {
     // The bridge code only has nGraph emitters for kOp-type nodes.
     if (node->type_ == NodeType::kOp) {
-      if (compiler_.ngraph_op_funcs_.count(node->operation_)) {
-        node->in_ngraph_ = true;
-
-        if (node->operation_ == "BatchNorm") {
-          auto shape = TShape_to_NShape(node->inputs_[0]->shape_);
-          if (shape[1] % 8 != 0) {
-            // MXNet outperforms nGraph in this case.
-            node->in_ngraph_ = false;
-          }
-        } else if (node->operation_ == "LeakyReLU") {
-          // We haven't yet implemented all activation functions for
-          // LeaklyReLU...
-          const std::string act_type =
-              get_default(node, "act_type", std::string("leaky"));
-          if (act_type != "leaky") {
-            node->in_ngraph_ = false;
-          }
-        }
+      if (compiler_.supported_ops.count(node->operation_)) {
+        node->in_ngraph_ = compiler_.supported_ops[node->operation_](node);
+        // nGraph doesn't yet support float16 or sparse storage.
+        auto bad_type = [](NodePtr node) {
```

Comment:
If you're trying to use lambdas less, would it make sense to convert this function to just being a static function in this .cc file?

Reply:
I'm not trying to use lambdas less, the point of moving functionality out of this file is mostly to support the Param ojects. Plus, accessing that static function requires templating, which requires a dynamic ptr cast, and that just gets nasty.